packages/mini-browser/src/browser/mini-browser-open-handler.ts

```diff
@@ -167,6 +167,14 @@ export class MiniBrowserOpenHandler extends NavigatableWidgetOpenHandler<MiniBro
         };
     }
 
+    protected canOpenSource(widget: Widget): boolean {
+        const extension = widget.title.label.toString().split('.').pop();
+        if (extension) {
+            return ['jpg', 'jpeg', 'png', 'bmp', 'gif', 'pdf'].indexOf(extension.toLocaleLowerCase()) === -1;
```

Comment:
Just to clarify, how did you determine the appropriate extension types to exclude?

Reply:
The types to exclude are those specified in `mini-browser-endpoint.js`, in which `ImageHandler` can support files with extensions 'jpg', 'jpeg', 'png', 'bmp', 'gif' and `PdfHandler` can support 'pdf'. Those are the types that can be previewed (*open with preview*) but the source is not to be shown, as #5110 mentioned.

Reply:
> The types to exclude are those specified in `mini-browser-endpoint.js`, in which `ImageHandler` can support files with extensions 'jpg', 'jpeg', 'png', 'bmp', 'gif' and `PdfHandler` can support 'pdf'. Those are the types that can be previewed (_open with preview_) but the source is not to be shown, as #5110 mentioned.

I'm not sure this is true, since we also define an `svg` handler but yet it can be previewed and it's source can be shown. The thing I don't understand is that all those formats do have multiple `open-handlers` namely code-editor and preview. From the code-editor we can trigger the preview, but from a preview we cannot go backwards and open the source.

Reply:
I'm just trying to think if there is a cleaner way to determine if the source can be opened for a uri without having to hardcode extensions.

Reply:
Yes you are right. I will try to figure out a non-hardcoding way to solve this.